yt_dlp/extractor/lrt.py

```diff
@@ -138,13 +137,15 @@ def _real_extract(self, url):
             'https://www.lrt.lt/radioteka/api/media', video_id,
             query={'url': f'/mediateka/irasas/{video_id}/{path}'})
 
-        return traverse_obj(media, {
-            'id': ('id', {int}, {str_or_none}),
-            'title': ('title', {str}),
-            'tags': ('tags', ..., 'name', {str}),
-            'categories': ('playlist_item', 'category', {str}, filter, all, filter),
-            'description': ('content', {clean_html}, {str}),
-            'timestamp': ('date', {lambda x: x.replace('.', '/')}, {unified_timestamp}),
-            'thumbnail': ('playlist_item', 'image', {urljoin('https://www.lrt.lt')}),
-            'formats': ('playlist_item', 'file', {lambda x: self._extract_m3u8_formats(x, video_id)}),
-        })
+        return {
+            'id': video_id,
+            **traverse_obj(media, {
+                'title': ('title', {str}),
+                'tags': ('tags', ..., 'name', {str}),
+                'categories': ('playlist_item', 'category', {str}, filter, all, filter),
+                'description': ('content', {clean_html}, {str}),
+                'timestamp': ('date', {lambda x: x.replace('.', '/')}, {unified_timestamp}),
+                'thumbnail': ('playlist_item', 'image', {urljoin('https://www.lrt.lt')}),
+                'formats': ('playlist_item', 'file', {lambda x: self._extract_m3u8_formats(x, video_id)}),
```

Comment:
if we're unpacking a traversal call now, we can pull `formats` out of there too
```suggestion
            'id': video_id,
            'formats': self._extract_m3u8_formats(media['playlist_item']['file'], video_id, 'mp4'),
            **traverse_obj(media, {
                'title': ('title', {str}),
                'tags': ('tags', ..., 'name', {str}),
                'categories': ('playlist_item', 'category', {str}, filter, all, filter),
                'description': ('content', {clean_html}, {str}),
                'timestamp': ('date', {lambda x: x.replace('.', '/')}, {unified_timestamp}),
                'thumbnail': ('playlist_item', 'image', {urljoin('https://www.lrt.lt')}),
```

Reply:
Shouldn't we be save with retrieving `media['playlist_item']['file']` in case the "playlist_item" key not found?

Reply:
no, `_extract_m3u8_formats` is already fatal (as it was before this change, too)

Reply:
Got it. Thanks for input!

Reply:
traversal tranformations are always called, even if the path is returning `None` before the transformation is reached. which is why i generally don't like using expensive methods in transformations like this

Reply:
Hmm, after extracting "formats", tests start to fail. Previously they didn't. I am not familiar with this code base as much as you, but it seems transformations were not called earlier.

Reply:
can you show the log

Reply:
and if we can't extract formats then we want to error. so something else is wrong if tests are failing because of formats

Reply:
Ah, sorry, that was a false alarm from my side. Different result was returned: mp4 instead m4a in test case. Will push updated changes in a min.